packages/core/src/components/GridLayout/GridLayout.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import css from './GridLayout.css';
+import PropTypes from 'prop-types';
+
+export default class GridLayout extends Component {
+  static propTypes = {
+    grid: PropTypes.arrayOf(
+      PropTypes.shape({
+        key: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+      }),
+    ).isRequired,
+    GridItemComponent: PropTypes.any,
+    limit: PropTypes.number,
+    col: PropTypes.number,
+    colGap: PropTypes.number,
+    rowHeight: PropTypes.number,
+  };
+
+  static defaultProps = {
+    grid: [],
+    GridItemComponent: 'div',
+    limit: 16,
+    col: 4,
+    colGap: 1,
+    rowHeight: 10
+  };
+
+  constructor(props) {
+    super(props);
+
+    const { grid, limit } = props;
+
+    this.state = {
+      grid: grid.slice(0, limit)
+    };
+  }
+
+  componentWillReceiveProps(props) {
+    const { grid, limit } = props;
+
+    this.setState({
+      grid: grid.slice(0, limit)
+    });
```

Comment:
are we just doing this for the sake of memoization? it's seems a bit of an antipattern using state to store some props, can just do this in the render method and we won't need state, this could then be a stateless component